netket/callbacks/invalidlossstopping.py

```diff
@@ -28,8 +28,8 @@ class InvalidLossStopping(struct.Pytree, mutable=True):
     """Number of epochs with invalid loss after which training will be stopped."""
 
     # caches
-    _last_valid_iter: int
-    """Last valid iteration, to check against patience"""
+    _invalid_steps: int
```

Comment:
Can you explain this change? 
The reason why we were storing the 'last valid iter' instead of 'invalid step' was because sometimes we do not call the callback at every step (if you specify step_size=2 in run for example).

So we need to store the interval..

Reply:
Ah, I think I understand. 
Is it because `step` was a float that is the time? and you wanted an integer?

Then you can use https://netket.readthedocs.io/en/latest/api/_generated/driver/netket.driver.AbstractVariationalDriver.html#netket.driver.AbstractVariationalDriver.step_count 

instead of step

Reply:
yep, that's why tdvp driver has both `driver.step_count` and `driver.step_value` . You can safely ignore the `step` argument passed to the callback (it will be identical to `driver.step_value` IIRC, and you can instead use `driver.step_count` which is an integer counter.

I was suggesting above to you step count because this will allow to not change the behaviour for VMC & similar optimisers, while you get what you want for tdvp (I think)